src/browser.js

```diff
@@ -67,51 +69,48 @@ export class FetchError {
 }
 
 // Returns a promise that will resolve to the content of the given path.
-export function fetchUrl(path, params, headers) {
-  return new Promise(function(resolve, reject) {
-    var xhr = new XMLHttpRequest();
-    if (params) {
-      var separator = '?';
-      params.forEach(function(param) {
-        var key = encodeURIComponent(String(param[0]));
-        var val = encodeURIComponent(String(param[1]));
-        path += separator + key + '=' + val;
-        separator = '&';
-      });
-    }
-    xhr.open('GET', path, true);
-    if (headers) {
-      utils.Map.wrap(headers).forEach(function(key, value) {
-        xhr.setRequestHeader(key, value);
-      });
-    }
-    xhr.onreadystatechange = function() {
-      if (xhr.readyState != 4)
-        return;
+export function fetchUrl(path, params, headers = {}) {
+  return new Promise(async function(resolve, reject) {
+    try {
+      if (params) {
+        var separator = '?';
+        params.forEach(function(param) {
+          var key = encodeURIComponent(String(param[0]));
+          var val = encodeURIComponent(String(param[1]));
+          path += separator + key + '=' + val;
+          separator = '&';
+        });
+      }
+
+      const response = await fetch(path, {headers});
```

Comment:
Replace `{headers}` by `headers`.

Reply:
that would not be correct?

`headers` is one of the valid keys used in a Fetch requests [optional second argument](https://fetch.spec.whatwg.org/#example-cors-safelisted-request-header-content-type). If the value of the `headers` variable is 

```json
{
  "pragma": "no-cache",
  "cache-control": "no-cache, must-revalidate",
}
```

Then the current code destructures to 

```js
await fetch(path, {"headers": {
  "pragma": "no-cache",
  "cache-control": "no-cache, must-revalidate",
}})
```

If I updated to your request, it would be 

```js
await fetch(path, {
  "pragma": "no-cache",
  "cache-control": "no-cache, must-revalidate",
})
```

which would not be passing any header through to the network

Reply:
You are right.

I did make local changes locally trying to disable CORS, and the error were coming from my changes, not from your patch.